article/_posts/2020-06-19-DL_philosophy.md

```diff
@@ -0,0 +1,54 @@
+
+* **Hierarchical feature composition**: CNNs work like visual processing has been thought to work in the mammalian ventral stream, by hierarchically composing more complex feature representations from simpler and less abstract ones. This is only a good **story**, but experts have vindicated the role of hierarchical feature composition by highlighting the increasing recoverability of more abstract features as one moves up the ventral stream hierarchy (i.e., feature space visualization).
+
+* **Systematic transformations of input to adjust for nuisance variation**: Sparse connectivity and regularization impose the assumption that the recognition of generalizable features should not depend on too many long-distance relations among features or subtle contextual details. During training, CNNs simply ignore hypotheses that violate these assumptions; but when the solution to a problem satisfies them, CNNs can find the solution more efficiently than more traditional neural networks. Thus, **perhaps** CNNs work so well because a wide class of classification and decision problems satisfy these assumptions. How far the utility of these assumptions extends into territory traditionally ascribed to "higher" cognition remains an open question. 
+
+* **Number of linear regions**: Another intuition pretending to explain the success of CNNs is the number of linear regions they can map in a problem's input space, relative to networks that do not possess their characteristic features. The ability to draw more distinct linear regions is advantageous because it allows neural networks to impose more complex boundaries between categories which are difficult to discriminate from one another because of many low-level perceptual similarities. This **is not a complete answer** to our why question because it still leaves room for further why questions, like why do CNNs learn these linear regions?
+
+![](/article/images/Philosophy_DL/feature_space.jpg)
+
+![](/article/images/Philosophy_DL/folding_ex.jpg)
+
+# Open Problems
+
+Most commentators agree that current deep learning methods fall short of implementing general intelligence, and it remains an open question as to whether some modification of current deep learning methods will be able to do so.
+
+* **CNNs and nativism/empiricism**: Nativism is a concept in psychology and philosophy which asserts certain concepts are "native" or in the brain at birth (given a priori). Empiricism is a theory that states that knowledge comes only or primarily from sensory experience. Philosophers think of cognition as ranging over four variables: algorithms, representational formats, innate knowledge and experience. A truly empiricist system should begin with nothing for representational format and innate knowledge, and only a very minimal amount of algorithms, deriving everything else from experience. Therefore, AlphaGo, AlphaStar etc. provide philosophical insight into empiricism.
```

Comment:
_Nativism is a concept in psychology and philosophy which asserts certain concepts are "native" or in the brain at birth (given a priori)_.  : is the `or` necessary?

Reply:
Addressed in bf1121e.